awscli/customizations/s3/s3handler.py

```diff
@@ -523,6 +525,80 @@ def _process_transfer(self, future):
             return CommandResult(1, 0)
 
 
+class S3TransferHandler(object):
+    def __init__(self, transfer_manager, cli_params, result_queue,
+                 result_recorder, result_processor):
+        """Backend for performing S3 transfers
+
+        :type transfer_manager: s3transfer.manager.TransferManager
+        :param transfer_manager: Transfer manager to use for transfers
+
+        :type cli_params: dict
+        :param cli_params: The parameters passed to the CLI command in the
+            form of a dictionary
+
+        :type result_queue: queue.Queue
+        :param result_queue: The queue to process results
+
+        :type result_recorder: ResultRecorder
+        :param result_recorder: Result recorder to get status of transfer
+
+        :type result_processor: ResultProcessor
+        :param result_processor: Result processor to process status updates
+            of transfer
+        """
+        self._transfer_manager = transfer_manager
+        self._result_queue = result_queue
```

Comment:
Yeah it felt a little weird that I was passing in a result recorder and a result processor that already had been built with a result recorder to the initializer. How does an interface like this sound?
```py
class ResultProcessorManager:
    def __init__(self, result_queue, result_recorder, result_processor):
        self.result_queue = result_queue
    def start(self):
    def shutdown(self):
    def get_final_result(self):
    def __enter__(self):
    def __exit__(self, exc_type, *args):
```
That way all I would have to pass in is this result processor manager to the s3 transfer handler. If I do this, should I make it a separate PR? I feel like that may be nicer.